### PR TITLE
Update dimension string when it is explicitly given in Pipeline @open sesame 10/20 18:25

### PR DESCRIPTION
**Pull request by again4you**
Even though the dimension string of the input tensor is given in the
pipeline, the `input` property is always fixed 4 dimensions (i.e.
d1:d2:d3:d4). This patch returns the dimension string considering tensor
dimension and rank count.

For example, given tensor filter parameters
'tensor_filter input=3:224:224', the returned dimension string is
`__3:224:224__` instead of `3:224:224:1`.

Signed-off-by: Sangjung Woo <sangjung.woo@samsung.com>

**Self evaluation:**
1. Build test: [*]Passed [ ]Failed []Skipped
2. Run test: [*]Passed [ ]Failed []Skipped

#### v2
* Rebase previous patches
* According to Jaeyun's advice, make a simple logic

#### v1
* first draft

### CODE REVIEW

**Review by myungjoo — APPROVED** [review]
LGTM

File: gst/nnstreamer/tensor_filter/tensor_filter_common.c [hunk 1]
```diff
@@ -266,6 +266,45 @@ gst_tensors_get_rank_string (const GstTensorFilterProperties * prop,
...
+
+    for (i = 0; i < tinfo->num_tensors; ++i) {
+      if (isInput)
+        dim_str =
+            gst_tensor_get_rank_dimension_string (tinfo->info[i].dimension,
+            prop->input_ranks[i]);
+      else
+        dim_str =
+            gst_tensor_get_rank_dimension_string (tinfo->info[i].dimension,
+            prop->output_ranks[i]);
```
Comment (jaeyun-jung) [inline gst/nnstreamer/tensor_filter/tensor_filter_common.c:295]: how about adding unsigned int * _rank?
may reduce code lines.
_rank = (isInput) ? prop->input_ranks : prop->output_ranks;

Comment (again4you) [reply gst/nnstreamer/tensor_filter/tensor_filter_common.c]: ACK!

**Comment by again4you** [comment]
@jaeyun-jung
Could you check the new patch and merge it?

**Review by jaeyun-jung — APPROVED** [review]

**Pull request merged**